src/main.ts

```diff
@@ -310,13 +311,22 @@ export class Purchases {
       `Presenting purchase form for package ${rcPackage.identifier}`,
     );
 
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
+      let brandingInfo: BrandingInfoResponse | null = null;
+      try {
+        brandingInfo = await this.backend.getBrandingInfo();
```

Comment:
We might want to display some sort of loading UI for this... It should be pretty fast though so maybe not needed?